sources/Load.php

```diff
@@ -2488,44 +2488,29 @@ function determineAvatar($profile)
 function detectServer()
 {
 	global $context;
+	static $server = null;
 
-	$context['server'] = array(
-		'is_iis'			=> serverIs('iis'),
-		'is_apache'			=> serverIs('apache'),
-		'is_litespeed' 		=> serverIs('litespeed'),
-		'is_lighttpd' 		=> serverIs('lighttpd'),
-		'is_nginx' 			=> serverIs('nginx'),
-		'is_cgi' 			=> serverIs('cgi'),
-		'is_windows' 		=> serverIs('windows'),
-		'iso_case_folding' 	=> serverIs('iso_case_folding'),
+	if ($server === null)
+	{
+		$server = new Server($_SERVER);
```

Comment:
I'd go for that ... just a ``` $server = new Server();``` and then pass the logic to the constructor, so same thing different location.
```
protected $__server = array()l;

public function __construct($server = null)
{
if (is_array($server))
$this->__server = $server;
else
$this->__server = $_SERVER;
}
```